dashboard/test/controllers/api/v1/pd/applications_controller_test.rb

```diff
@@ -403,6 +403,81 @@ class ApplicationsControllerTest < ::ActionController::TestCase
       assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
     end
 
+    test 'update appends to timestamp log if workshop admin changes application from unlocked to locked' do
+      sign_in @workshop_admin
+      @csp_facilitator_application.update(status_timestamp_change_log: '[]', locked_at: nil)
+      @csp_facilitator_application.reload
+
+      assert_equal [], @csp_facilitator_application.sanitize_status_timestamp_change_log
+      assert_equal false, @csp_facilitator_application.locked?
+
+      # Changing application from unlocked to locked
+      post :update, params: {id: @csp_facilitator_application.id, application: {status: @csp_facilitator_application.status, locked: true}}
+      @csp_facilitator_application.reload
+
+      expected_log = [
+        {
+          title: 'Application is locked',
+          changing_user_id: @workshop_admin.id,
+          changing_user_name: @workshop_admin.name,
+          time: Time.zone.now
+        }
+      ]
+
+      assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
+
+      # Setting application to locked again
+      post :update, params: {id: @csp_facilitator_application.id, application: {status: @csp_facilitator_application.status, locked: true}}
+      @csp_facilitator_application.reload
+
+      assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
+    end
+
+    test 'update appends to timestamp log if workshop admin changes application from locked to unlocked' do
+      sign_in @workshop_admin
+      @csp_facilitator_application.update(status_timestamp_change_log: '[]', locked_at: Time.zone.now)
+      @csp_facilitator_application.reload
+
+      assert_equal [], @csp_facilitator_application.sanitize_status_timestamp_change_log
+      assert_equal true, @csp_facilitator_application.locked?
+
+      # Changing application from locked to unlocked
+      post :update, params: {id: @csp_facilitator_application.id, application: {status: @csp_facilitator_application.status, locked: false}}
+      @csp_facilitator_application.reload
+
+      expected_log = [
+        {
+          title: 'Application is unlocked',
+          changing_user_id: @workshop_admin.id,
+          changing_user_name: @workshop_admin.name,
+          time: Time.zone.now
+        }
+      ]
+
+      assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
+
+      # Setting application to unlocked again
+      post :update, params: {id: @csp_facilitator_application.id, application: {status: @csp_facilitator_application.status, locked: false}}
+      @csp_facilitator_application.reload
+
+      assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
+    end
+
+    test 'update does not append to timestamp log if application lock is not changed by workshop admin' do
```

Comment:
this feels like it is testing our permissions rather than the behavior of the log - I would change this to verify that it does not append to the log if the locked status is not changed. So log in the admin and have them try and unlock an already-unlocked application, and verify that it doesn't log it.

Reply:
Agree, it's more of testing permission here. I can remove it. The part that tests logging doesn't change when application lock state stays the same is embedded in other 2 tests.